### PR TITLE
Xfsprogs update1

### PR DESCRIPTION
**Pull request by AZaugg**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 
Updating xfsprogs to version 5.15 to match kernel version. In addition
adding a new package called inih, which is a new C library that xfsprogs

requires###### Change Log  
Updated xfsprogs to 5.15
Pull in new package which is required by xfsprogs

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Ran the command sudo make build-packages -j$(nproc) REBUILD_TOOLS=y SRPM_PACK_LIST="xfsprogs"
- Verified that package build
```
azaugg@azaugg-ld77 [ ~/CBL-Mariner ]$ sudo rpm -qlp /home/azaugg/CBL-Mariner/out/RPMS/x86_64/xfsprogs-5.15.0-1.cm2.x86_64.rpm
/lib64/libhandle.so.1.0.3
/sbin/fsck.xfs
/sbin/mkfs.xfs
/sbin/xfs_repair
/usr/lib64/xfsprogs/xfs_scrub_all.cron
/usr/sbin/xfs_admin
...
```

- Checksum test
```
azaugg@azaugg-ld77 [ ~ ]$ sha256sum r57.tar.gz
f03f98ca35c3adb56b2358573c8d3eda319ccd5287243d691e724b7eafa970b3  r57.tar.gz
azaugg@azaugg-ld77 [ ~ ]$ sha256sum xfsprogs-5.15.0.tar.xz
13d2bf3830eba0027ab558825439dd521f61c7266783d7917213432872e74d56  xfsprogs-5.15.0.tar.xz
```

### CODE REVIEW

File: SPECS/inih/inih.spec [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+%{_libdir}/pkgconfig/inih.pc
+%{_libdir}/pkgconfig/INIReader.pc
+%{_libdir}/lib%{name}.so
+%{_libdir}/libINIReader.so
+
+
+%changelog
+* Fri Jul 28 2023 Andy Zaugg <azaugg@linkedin.com> - 57-1
+- Updated version to 57
+- Initial CBL-Mariner import from Fedora (license: BSD)
```
Comment (mfrw) [inline SPECS/inih/inih.spec:70]: I think this comment here corresponds to the license of the `spec` file which IMO is `MIT`.

The comment should probably read something like:

`Initial CBL-Mariner import from Fedora 36 (license: MIT)`
and substitute the appropriate version of Fedora where we imported it from.

**Comment by dmcilvaney** [comment]
https://dev.azure.com/mariner-org/mariner/_build/results?buildId=405634&view=results

**Comment by dmcilvaney** [comment]
> https://dev.azure.com/mariner-org/mariner/_build/results?buildId=405634&view=results

Oops, forgot to grab the new xfsprogs sources. Re-running

**Comment by dmcilvaney** [comment]
> > https://dev.azure.com/mariner-org/mariner/_build/results?buildId=405634&view=results
> 
> Oops, forgot to grab the new xfsprogs sources. Re-running

Looks like a stage in the pipeline failed, but the builds themselves seem good.

File: SPECS/inih/inih.spec (hunk 1 above)
Comment (AZaugg) [reply SPECS/inih/inih.spec]: Ah, thank you. Updated the Fedora version

File: SPECS/inih/inih.spec [hunk 2]
```diff
@@ -0,0 +1,135 @@
+Summary:  A Simple INI file parser library
+Name:     inih
+Version:  57
+Release:  1%{?dist}
+License:  BSD
+URL:      https://github.com/benhoyt/inih
+#Source0:  https://github.com/benhoyt/inih/archive/refs/tags/r57.tar.gz
+Source0:  inih-r57.tar.gz
```
Comment (dmcilvaney) [inline SPECS/inih/inih.spec:8]:
```suggestion
Source0:  https://github.com/benhoyt/inih/archive/refs/tags/r57.tar.gz#/inih-r57.tar.gz
```

**Review by dmcilvaney — CHANGES_REQUESTED** [review]

File: SPECS/inih/inih.spec (hunk 2 above)
Comment (AZaugg) [reply SPECS/inih/inih.spec]: Done

**Comment by dmcilvaney** [comment]
https://dev.azure.com/mariner-org/mariner/_build/results?buildId=411778&view=results

**Comment by dmcilvaney** [comment]
https://dev.azure.com/mariner-org/mariner/_build/results?buildId=411858&view=results

File: SPECS/xfsprogs/xfsprogs.spec [hunk 3]
```diff
@@ -84,6 +88,9 @@ make -k check |& tee %{_specdir}/%{name}-check-log || %{nocheck}
 %defattr(-,root,root)
 
 %changelog
+* Fri Dec 28 2023 Andy Zaugg <azaugg@linkedin.com> - 5.15-1
```
Comment (dmcilvaney) [inline SPECS/xfsprogs/xfsprogs.spec:91]:
```suggestion
* Fri Jul 28 2023 Andy Zaugg <azaugg@linkedin.com> - 5.15-1
```

**Review by dmcilvaney — CHANGES_REQUESTED** [review]

**Review by dmcilvaney — APPROVED** [review]

**Pull request merged**